packages/plugin-dcap/src/actions/on-chain.ts

```diff
@@ -0,0 +1,111 @@
+import type { Action } from "@elizaos/core";
+import { verifyAndAttestOnChain } from "../dcap.js";
+import { getQuote } from "../quote.js";
+import { DCAPMode } from "../types.js";
+import {
+    getDCAPMode,
+    getTEEMode,
+    hasPrivateKey,
+    hasTEEMode,
+} from "../utils.js";
+
+export const dcapOnChainVerifyAction: Action = {
+    name: "DCAP_ON_CHAIN",
+    description:
+        "This plugin is used to generate DCAP attestation and verify it on-chain. The user can also use the keyword DCAP_ON_CHAIN to trigger this action.",
+    similes: [
+        "DCAP",
+        "DCAP_ATTESTATION",
+        "DCAP_TEE",
+        "DCAP_SGX",
+        "DCAP_TDX",
+        "VERIFY_ATTESTATION",
+        "VERIFY_DCAP",
+        "DCAP_VERIFICATION",
+        "ATTESTATION",
+        "GENERATE_ATTESTATION",
+    ],
+    examples: [
+        [
+            {
+                user: "{{user1}}",
+                content: {
+                    text: "Generate a DCAP attestation and verify it on-chain",
```

Comment:
Move the action to this part or you'll get some odd behavior. At least this is what I ran into when interacting with Twitter when requesting a remote attestation from an agent.

Reply:
```suggestion
                    text: "Generate a DCAP attestation and verify it on-chain",
                    action: "DCAP_ON_CHAIN",
```

Reply:
Hi @HashWarlock , thanks and already made the change according to your suggestion!